x-pack/plugins/fleet/public/applications/fleet/sections/settings/components/fleet_server_hosts_flyout/use_fleet_server_host_form.tsx

```diff
@@ -77,7 +77,16 @@ function validateFleetServerHosts(value: string[]) {
   const res: Array<{ message: string; index: number }> = [];
   const hostIndexes: { [key: string]: number[] } = {};
   value.forEach((val, idx) => {
-    if (!val.match(URL_REGEX)) {
+     if(val === ''){
+       return [
```

Comment:
I think the problem is that you return from a `forEach` function, which doesn't return from the validate function. To fix it, you can use `res.push` like in the other branches:
```
 if(!val){
      res.push({
          message: i18n.translate('xpack.fleet.settings.fleetServerHostsEmptyError', {
            defaultMessage: 'Host URL is required',
          }),
          index: idx
        },
      );
    }
   else if (!val.match(URL_REGEX)) {
```

Reply:
Thank you for the feedback. I made changes based on this.